### PR TITLE
Add button platform to Roborock

### PR DESCRIPTION
**Pull request by Lash-L**
## Proposed change

adds 4 new buttons that can reset the consumables/ maintenance items.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/103010
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/roborock.markdown [hunk 1]
```diff
@@ -112,6 +114,18 @@ Do not disturb - This enables _Do not disturb_ during the time frame you have se
 
 Volume - This allows you to control the volume of the robot's voice. For example, when it states "Starting cleaning". This allows you to set the volume to 0%, while the app limits it to 20%.
 
+### Button
+
+There are currently four buttons that allow you to reset the various maintenance items on your vacuum. This cannot be undone after pressing - so they are disabled by default to make sure you know what you are doing before you click on them.
```
Comment (c0ffeeca7) [inline source/_integrations/roborock.markdown]:
```suggestion
There are currently four buttons that allow you to reset the various maintenance items on your vacuum. Pressing the button cannot be undone. For this reason,  they are disabled by default to make sure they are not pressed unintentionally.
```

File: source/_integrations/roborock.markdown [hunk 2]
```diff
@@ -112,6 +114,18 @@ Do not disturb - This enables _Do not disturb_ during the time frame you have se
...
+
+Reset sensor consumable - The sensors on your vacuum are intended to be cleaned after 30 hours of use
```
Comment (c0ffeeca7) [inline source/_integrations/roborock.markdown]:
```suggestion
Reset sensor consumable - The sensors on your vacuum are expected to be cleaned after 30 hours of use.
```

File: source/_integrations/roborock.markdown [hunk 3]
```diff
@@ -112,6 +114,18 @@ Do not disturb - This enables _Do not disturb_ during the time frame you have se
...
+
+Reset side brush consumable - The side brush is expected to be replaced every 200 hours.
+
+Reset main brush consumable - the Main brush/ roller is expected ot be replaced every 300 hours.
```
Comment (c0ffeeca7) [inline source/_integrations/roborock.markdown]:
```suggestion
Reset main brush consumable - The main brush/ roller is expected to be replaced every 300 hours.
```

File: source/_integrations/roborock.markdown [hunk 4]
```diff
@@ -112,6 +114,18 @@ Do not disturb - This enables _Do not disturb_ during the time frame you have se
...
+
+Reset air filter - the Air filter is expected to be replaced every 150 hours.
```
Comment (c0ffeeca7) [inline source/_integrations/roborock.markdown]:
```suggestion
Reset air filter - The Air filter is expected to be replaced every 150 hours.
```

File: source/_integrations/roborock.markdown [hunk 5]
```diff
@@ -112,6 +114,18 @@ Do not disturb - This enables _Do not disturb_ during the time frame you have se
...
+
+There are currently four buttons that allow you to reset the various maintenance items on your vacuum. Pressing the button cannot be undone. For this reason,  they are disabled by default to make sure they are not pressed unintentionally.
+
+Reset sensor consumable - The sensors on your vacuum are expected to be cleaned after 30 hours of use.
+
+Reset side brush consumable - The side brush is expected to be replaced every 200 hours.
+
+Reset main brush consumable - The main brush/ roller is expected to be replaced every 300 hours.
+
+Reset air filter - The Air filter is expected to be replaced every 150 hours.
```
Comment (c0ffeeca7) [inline source/_integrations/roborock.markdown]:
```suggestion
Reset air filter - The air filter is expected to be replaced every 150 hours.
```

File: source/_integrations/roborock.markdown [hunk 6]
```diff
@@ -112,6 +114,18 @@ Do not disturb - This enables _Do not disturb_ during the time frame you have se
 
 Volume - This allows you to control the volume of the robot's voice. For example, when it states "Starting cleaning". This allows you to set the volume to 0%, while the app limits it to 20%.
 
+### Button
+
+There are currently four buttons that allow you to reset the various maintenance items on your vacuum. Pressing the button cannot be undone. For this reason,  they are disabled by default to make sure they are not pressed unintentionally.
```
Comment (c0ffeeca7) [inline source/_integrations/roborock.markdown]:
```suggestion
There are currently four buttons that allow you to reset the various maintenance items on your vacuum. Pressing the button cannot be undone. For this reason, the buttons are disabled by default to make sure they are not pressed unintentionally.
```

File: source/_integrations/roborock.markdown [hunk 7]
```diff
@@ -112,6 +114,18 @@ Do not disturb - This enables _Do not disturb_ during the time frame you have se
...
+There are currently four buttons that allow you to reset the various maintenance items on your vacuum. Pressing the button cannot be undone. For this reason, the buttons are disabled by default to make sure they are not pressed unintentionally.
+
+Reset sensor consumable - The sensors on your vacuum are expected to be cleaned after 30 hours of use.
+
+Reset side brush consumable - The side brush is expected to be replaced every 200 hours.
+
+Reset main brush consumable - The main brush/ roller is expected to be replaced every 300 hours.
+
+Reset air filter - The air filter is expected to be replaced every 150 hours.
+
```
Comment (c0ffeeca7) [inline source/_integrations/roborock.markdown:128]: just to be sure I understand this correctly: pressing the reset button means the consumable was replaced / cleaned?

Comment (Lash-L) [reply source/_integrations/roborock.markdown:128]: Yep!

**Review by c0ffeeca7 — DISMISSED** [review]
Thank you, @Lash-L :+1:

File: source/_integrations/roborock.markdown [hunk 8]
```diff
@@ -9,6 +9,7 @@ ha_category:
   - Switch
   - Time
   - Vacuum
+  - Button
```
Comment (gjohansson-ST) [inline source/_integrations/roborock.markdown]: Sorting

File: source/_integrations/roborock.markdown [hunk 9]
```diff
@@ -25,6 +26,7 @@ ha_platforms:
   - switch
   - time
   - vacuum
+  - button
```
Comment (gjohansson-ST) [inline source/_integrations/roborock.markdown]: Sorting

**Review by gjohansson-ST — CHANGES_REQUESTED** [review]

**Review by gjohansson-ST — APPROVED** [review]
Thanks @Lash-L 👍

**Pull request merged**